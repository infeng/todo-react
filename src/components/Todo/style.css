.todo {
  padding: 10px;
  color: black;
  -webkit-transition: color .3s ease;
  transition: color .3s ease;   
}
.todo.notlast {
  border-bottom: 1px solid #ededed;
}
.todo.completed {
  color: #d9d9d9;
}