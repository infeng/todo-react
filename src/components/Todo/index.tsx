import * as React from 'react';
import './style.css';

interface Props {
  onClick: (e: any) => void;
  text: string;
  completed: boolean;
  isLast: boolean;
}

export default class Todo extends React.Component<Props, any> {
  render() {
    let className: string = 'todo';
    if (!this.props.isLast) {
      className += ' notlast';
    }
    if (this.props.completed) {
      className += ' completed';
    }
    return (
      <li className={className}
      onClick={this.props.onClick}
      style={{
        textDecoration: this.props.completed ? 'line-through' : 'none',
        cursor: this.props.completed ? 'default' : 'pointer',
      }}
      >
        {this.props.text}
      </li>
    );
  }
}
