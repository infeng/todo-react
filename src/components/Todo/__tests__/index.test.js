import React from 'react';
import { mount, render } from 'enzyme';
import Todo from '..';

function createTodo(props) {
  return (
    <Todo text="hello world" {...props}/>
  );
}

describe('Todo', () => {
  it('render Todo correctly', () => {
    let todo = createTodo();

    const wrapper = render(todo);
    expect(wrapper).toMatchSnapshot();

    const wrapper2 = mount(todo);
    expect(wrapper2.text()).toBe('hello world');
  });

  it('completed Todo', () => {
    let todo = createTodo({completed: true});

    const wrapper = mount(todo);
    expect(wrapper.hasClass('completed')).toBe(true);
  });

  it('last Todo', () => {
    let todo = createTodo({isLast: true});

    const wrapper = mount(todo);
    expect(wrapper.hasClass('notlast')).toBe(false);
  });
});
