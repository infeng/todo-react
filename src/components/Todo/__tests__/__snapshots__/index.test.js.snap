// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo render Todo correctly 1`] = `
<li
  class="todo notlast"
>
  hello world
</li>
`;
